src/run.ts

```diff
@@ -95,11 +97,51 @@ function getDockerSecretArguments(secretName: string): string[] {
 
 function getGenericSecretArguments(secretName: string): string[] {
     const secretArguments = core.getInput('arguments');
+    const parsedArgument = fromLiteralsToFromFile(secretArguments);
     let args = ['create', 'secret', 'generic', secretName];
-    args.push(...argStringToArray(secretArguments));
+    args.push(...argStringToArray(parsedArgument));
     return args;
 }
 
+/**
+ * Takes a valid kubectl arguments and parses --from-literal to --from-file
+ * @param secretArguments 
+ */
+function fromLiteralsToFromFile(secretArguments: string): string {
+    const parsedArgument = secretArguments.split("--").reduce((argumentsBuilder, argument) => {
+        if (argument && !argument.startsWith("from-literal=")) {
+            argumentsBuilder += " --" + argument;
+        } else if (argument && argument.startsWith("from-literal=")) {
+            const command = argument.substring("from-literal=".length);
+            /* The command starting after 'from-literal=' contanis a 'key=value' format. The secret itself might contain a '=', 
+            Hence the substring than a split*/
+            if (command.indexOf("=") == -1) throw new Error('Invalid from-literal input. It should contain a key and value');
+            const secretName = command.substring(0, command.indexOf("="));
+            const secretValue = command.substring(command.indexOf("=") + 1);
+            argumentsBuilder += " --from-file=" + createFile(secretName.trim(), secretValue, true);
+        }
+        return argumentsBuilder;
+    });
+    return parsedArgument;
+}
+
+/**
+ * 
+ * @param fileName The fileName in case of a file needs to be created in TEMP folder else the entire filepath
+ * @param data File data
+ * @param isTempFile Boolean to indicate if file needs to be created in TEMP folder
+ */
+function createFile(fileName: string, data: string, isTempFile: boolean): string {
```

Comment:
I like this convention. It seems cleaner! 😃 
But instead of `isTempFile` call it `inTempFolder` maybe?